packages/frontend/src/components/accounts/ledger/SignInLedgerWrapper.js

```diff
@@ -106,73 +111,84 @@ export function SignInLedger(props) {
         dispatch(clearAccountState());
     };
 
-    const onClose = () => {
-        Mixpanel.track('IE-Ledger Close ledger confirmation');
+    const handleContinue = () => {
+        dispatch(redirectToApp());
+    };
+
+    const handleCancel = () => {
+        dispatch(clearSignInWithLedgerModalState());
+    };
+
+    const activeView = () => {
+        if (!signInWithLedgerStatus) {
+            return VIEWS.AUTHORIZE;
+        }
         if (signInWithLedgerStatus === LEDGER_MODAL_STATUS.CONFIRM_PUBLIC_KEY) {
-            controllerHelperApi.abort();
+            return VIEWS.SIGN_IN;
         }
         if (signInWithLedgerStatus === LEDGER_MODAL_STATUS.ENTER_ACCOUNTID) {
-            dispatch(clearSignInWithLedgerModalState());
+            return VIEWS.ENTER_ACCOUNT_ID;
+        }
+        if (signInWithLedgerStatus === LEDGER_MODAL_STATUS.CONFIRM_ACCOUNTS || signInWithLedgerStatus === LEDGER_MODAL_STATUS.SUCCESS) {
+            return VIEWS.IMPORT_ACCOUNTS;
+        }
+    };
+
+    const getCurrentViewComponent = (view) => {
+        switch (view) {
+            case VIEWS.AUTHORIZE:
+                return (
+                    <Authorize
```

Comment:
We're not passing in the `props` prop here, does that prevent the cancel button from working in the `Authorize` component?

Reply:
Thanks for catching it, I was planning to do it differently. Changed it to `redirectTo` action. It dispatches `push` from `connected-react-router` and makes redux devtool work properly.